code/lib/instrumenter/src/instrumenter.ts

```diff
@@ -491,15 +493,15 @@ export class Instrumenter {
         if (typeof arg !== 'function' || Object.keys(arg).length) return arg;
 
         return (...args: any) => {
-          // Set the cursor and parentId for calls that happen inside the callback.
-          const { cursor, parentId } = this.getState(call.storyId);
-          this.setState(call.storyId, { cursor: 0, parentId: call.id });
-          const restore = () => this.setState(call.storyId, { cursor, parentId });
+          // Set the cursor and ancestors for calls that happen inside the callback.
+          const { cursor, ancestors } = this.getState(call.storyId);
+          this.setState(call.storyId, { cursor: 0, ancestors: [...ancestors, call.id] });
+          const restore = () => this.setState(call.storyId, { cursor, ancestors });
 
           // Invoke the actual callback function.
           const res = arg(...args);
 
-          // Reset cursor and parentId to their original values before we entered the callback.
+          // Reset cursor and ancestors to their original values before we entered the callback.
           if (res instanceof Promise) res.then(restore, restore);
```

Comment:
Should we be returning `res.then(...)` rather than just hooking it up and not doing anything with it?

Reply:
We are returning `res` 2 lines down. It'll still have those callbacks attached.

Reply:
It'll have the callbacks attached but a caller will not wait on the callbacks running. There's no guarantee the callbacks will run before the promise resolves. I guess it is not an issue but just a little detail that might trip something up down the track.